File: src/Model/Application/Split.cs
namespace Model.Application
{
    public class Split
    {
        public int SplitId { get; set; }
        public string SplitName { get; set; }
        public string SplitDescription { get; set; }
        public decimal SplitValue { get; set; }
        public int CategoryId { get; set; }
        public string CategoryName { get; set; }
        public string CategoryDescription { get; set; }
        public int AccountSourceId { get; set; }
        public string AccountSourceName { get; set; }
        public string AccountSourceDescription { get; set; }
        public int TransferId { get; set; }
        public string TransferName { get; set; }
        public string TransferDescription { get; set; }
        public DateTime TransferDate { get; set; }

        public int OrderId { get; set; }
        public decimal Percentage { get; set; }

        public string TransferDateFormated => TransferDate.ToString("yyyy-MM-dd HH:mm");
        public string SplitValueFormated => SplitValue.ToString("N2");
    }
}

File: src/budgetwallettsapp/src/types/split.ts
import {Account} from "./account";
import {Category} from "./category";

export interface Split {
    splitId: number;
    splitName: string;
    splitDescription: string;
    splitValue: number;
    categoryId: number;
    categoryName: string;
    categoryDescription: string;
    accountSourceId: number;
    accountSourceName: string;
    accountSourceDescription: string;
    transferId: number;
    transferName: string;
    transferDescription: string;
    transferValue: number;
    transferDate: Date;
    isActive: boolean;
    orderId: number ;
    percentage: number;
    isShaded: boolean | null;
    transferDateFormated: string;
    splitValueFormated: string ;
    transferValueFormated: string;
  }
  
  export interface Transfer {
    transferId: number;
    transferName: string;
    transferDescription: string;
    transferValue: number;
    transferDate: string;
    budgetId: number;
    accountSourceId: number;
    accountSourceName: string;
    accountSourceDescription: string;
    splits: Split[];
    accounts: Account[];
    categories: Category[];
  }
  
  export interface SplitChartItem {
    periodOrderId: number;
    budgetPartSumValue: number;
    spltPartSumValue: number;
  }
  
  export interface SplitResponse {
    splits: Split[];
    splitSummary: {
      budgetValue: number;
    };
    splitChartItems: SplitChartItem[];
  }
  

  export interface ChartData {
    xAxis: { data: number[] }[];
    series: {
      showMark: boolean;
      name: string;
      data: number[];
      valueFormatter?: (value: number | null) => string;
    }[];
  }
  